### PR TITLE
StevenJohnson12

### PR DESCRIPTION
**Pull request by StevenJohnson12**
## Describe your project in 1-3 sentences

I am building an automated door alert system for my home which sends an alert SMS to my phone whenever a door is opened

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Comment by wardecs** [comment]
Honestly, this seems like a really cool idea. I'm interested to hear more! Keep me updated when you are done, i'd really like to see how this one turns out.

File: StevenJohnson12 [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+5. Configure the arduino uno (already own) to recieve this signal sent by the flipper via an antenna.
+6. Set up the software on the arduino to run a program sending a request to the twilio API to send an SMS message to my phone saying "Front door opened at: TIME"
+
+## Budget
+
+What materials will you need for your project? Where will you get them? How much does it cost? Please include all materials, including components you already own. Make sure to factor in shipping costs and sales tax.
+
+| Product         | Supplier/Link                         | Cost   |
+| --------------- | ------------------------------------- | ------ |
+| Flipper Zero Device   | https://shop.flipperzero.one/ | $169.00 + $7.40 shipping cost  |
```
Comment (cfanoulis) [inline StevenJohnson12:28]: Hi, it's very cool you're trying to use a flipper zero :) any reason you chose that over a simple microcontroller with a distance breakout board?

File: StevenJohnson12 [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Door security system
+
+## Summary
+
+My home has been broken into 2 times in the last year when I was not home. I don't currently have a proper security system installed for my doors & windows, because my parents are unable to afford one. So for my project I plan on creating an automatic wireless alert system for my front door using a Flipper Zero device to receive input from a door sensor & then use the radio frequency transmitter on the flipper to send out a signal to my Arduino Uno with a radio receiver antenna connected to my wifi network, running an automatic notification system to automatically send me a text message to my phone using the twilio API. I will then be notified by text every time the door is opened when i'm not home and hopefully this will prevent and future break-ins. 
+
+## Plan
+
+What steps are you going to take to build it? What tools are you going to use? What will you do first, second, third, etc.?
+1. Buy all required items listed in the budget. I already own most of the items needed for this project (connectors, breadboards, many other arduino components.
```
Comment (cfanoulis) [inline StevenJohnson12:15]: please include all the items in your budget (with a price of 0$), as we want to keep an accurate bill of materials

File: StevenJohnson12 [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+| --------------- | ------------------------------------- | ------ |
+| Flipper Zero Device   | https://shop.flipperzero.one/ | $169.00 + $7.40 shipping cost  |
+| WIFI Dev Board | https://shop.flipperzero.one/products/wifi-devboard  | $29.00  + $7.40 shipping cost |
+| Prototyping Boards   | https://shop.flipperzero.one/products/proto-boards | $10.00 + $4.10 shipping cost  |
+| Arduino Uno   | https://store-usa.arduino.cc/products/arduino-uno-rev3?selectedStore=us | $0.00 (Already Own)  |
+| Cables   | ALREADY OWN | $0.00  |
+
+
+
+| Total           |                                       | $226.90 |
```
Comment (cfanoulis) [inline StevenJohnson12:36]: I see you have noted shipping seperately, is it included in your total?
Also, have you accounted for sales taxes?

**Review by cfanoulis — CHANGES_REQUESTED** [review]
Hello Steven! Thanks for your proposal! I've commented on some few things on your budget/bill of materials. Please address those before we can move ahead :)

File: StevenJohnson12 [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+| Product         | Supplier/Link                         | Cost   |
+| --------------- | ------------------------------------- | ------ |
+| Flipper Zero Device   | https://shop.flipperzero.one/ | $169.00 + $4.50 shipping cost & $10.70 in taxes  |
+| WIFI Dev Board | https://shop.flipperzero.one/products/wifi-devboard  | $29.00  + $3.60 shipping cost & $1.84 in taxes |
+| Prototyping Boards   | https://shop.flipperzero.one/products/proto-boards | $10.00 + $4.10 shipping cost & $0.60 in taxes |
+| Arduino Uno   | https://store-usa.arduino.cc/products/arduino-uno-rev3?selectedStore=us | $0.00 (Already Own)  |
+| Cables   | ALREADY OWN | $0.00  |
+| Distance Sensor   | ALREADY OWN | $0.00  |
+| 2x Antenna Reciever/Transmitter   | ALREADY OWN | $0.00  |
```
Comment (cfanoulis) [inline StevenJohnson12:34]: can you please detail what kind of antenna you're using?

File: StevenJohnson12 [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+What materials will you need for your project? Where will you get them? How much does it cost? Please include all materials, including components you already own. Make sure to factor in shipping costs and sales tax.
+
+| Product         | Supplier/Link                         | Cost   |
+| --------------- | ------------------------------------- | ------ |
+| Flipper Zero Device   | https://shop.flipperzero.one/ | $169.00 + $4.50 shipping cost & $10.70 in taxes  |
+| WIFI Dev Board | https://shop.flipperzero.one/products/wifi-devboard  | $29.00  + $3.60 shipping cost & $1.84 in taxes |
+| Prototyping Boards   | https://shop.flipperzero.one/products/proto-boards | $10.00 + $4.10 shipping cost & $0.60 in taxes |
+| Arduino Uno   | https://store-usa.arduino.cc/products/arduino-uno-rev3?selectedStore=us | $0.00 (Already Own)  |
+| Cables   | ALREADY OWN | $0.00  |
+| Distance Sensor   | ALREADY OWN | $0.00  |
```
Comment (cfanoulis) [inline StevenJohnson12:33]: also please include what kind of sensor :)

File: StevenJohnson12 [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+| 2x Antenna Reciever/Transmitter   | ALREADY OWN | $0.00  |
+| Saudering Equipment   | ALREADY OWN | $0.00  |
```
Comment (cfanoulis) [inline StevenJohnson12:35]:
```suggestion
| Soldering Equipment   | ALREADY OWN | $0.00  |
```
also, I assume you already have solder/wick/etc? if so, can you include that as well?

**Review by cfanoulis — CHANGES_REQUESTED** [review]
new review, new questions :)

**Review by cfanoulis — COMMENTED** [review]
thanks for your proposal :) we're almost done (i swear)

do you plan on making/buying a case to hold your project? don't see anything in your budget 🔍

**Review by cfanoulis — APPROVED** [review]
lovin' the story behind this project, ngl. let's get you a grant - someone from @hackclub/bank-ops will be in touch to get you setup on Hack bank :)

**Review by MelSmith104 — COMMENTED** [review]
hey @StevenJohnson12 :) just updating it here that we're already chatting on slack!

**Review by MelSmith104 — APPROVED** [review]

**Pull request merged**